rotkehlchen/db/filtering.py

```diff
@@ -787,3 +793,12 @@ def prepare(self) -> Tuple[List[str], List[Any]]:
         filters = [f'counterparty IN ({questionmarks})']
         bindings = self.protocols
         return filters, bindings
+
+
+@dataclass(init=True, repr=True, eq=True, order=False, unsafe_hash=False, frozen=False)
+class DBIgnoredAssetsFilter(DBFilter):
+    asset_key: str
+
+    def prepare(self) -> Tuple[List[str], List[Any]]:
+        filters = [f'{self.asset_key} NOT IN (SELECT value FROM multisettings WHERE name="ignored_asset")']  # noqa: E501
```

Comment:
We probably need to find a way to combine precomputed sql values in the DB queries and filters. This will be doing `SELECT value FROM multisettings WHERE name="ignored_asset"` for every single event, which will be probably slow.

Reply:
As I know, sql is a declarative language which means that it automatically optimizes queries.

I've written this [script](https://gist.github.com/nebolax/323b03e2b109d382ee1d4c26e523d47a) to compare different queries. Queries are written differently but do the same thing. The results are as follows:

`formatted_query_time = 7.0457764019999995` (formatted with python f-strings)
`sql_query_time = 7.462006656999998` (just sql queries, no formatting)
`prepared_query_time = 7.749979786` (questionmarks + parameters)

Results are pretty close, aren't they? So I would say that query's syntax doesn't play a big role. What do you think?

Reply:
Closing after discussion in Discord. So no change is needed here